cmd/catalyst-node/catalyst-node.go

```diff
@@ -434,25 +444,47 @@ func redirectHlsHandler() http.Handler {
 		lat := r.Header.Get("X-Latitude")
 		lon := r.Header.Get("X-Longitude")
 
-		nodeAddr, err := getClosestNode(playbackID, lat, lon)
-		if err != nil {
+		var nodeAddr, validPrefix string
+		var err error
+
+		for _, prefix := range redirectPrefixes {
```

Comment:
I think we should probably parallelize this with a waitGroup — otherwise the `videorec` has to wait for a round-trip for `video` every single time